File: DemoObjCApp/Customization/CustomNotificationCenter.m
//
//  CustomNotificationCenter.m
//  DemoObjcApp
//
//  Created by Jae Kim on 1/15/19.
//  Copyright © 2019 Optimizely. All rights reserved.
//

#import "CustomNotificationCenter.h"
@import Optimizely;

@implementation CustomNotificationCenter

-(instancetype)init {
    self = [super init];
    if (self != nil) {
        _notificationId = 0;
    }
    
    return self;
}

@end

File: DemoObjCApp/Customization/CustomNotificationCenter.swift
//
//  CustomNotificationCenter.swift
//  DemoSwiftiOS
//
//  Created by Jae Kim on 1/11/19.
//  Copyright © 2019 Optimizely. All rights reserved.
//

import Foundation
import OptimizelySwiftSDK

typealias OPTNotificationCenter = OptimizelySwiftSDK.OPTNotificationCenter

class CustomNotificationCenter: OPTNotificationCenter {
    var notificationId: Int
    
    init() {
        notificationId = 0
    }
    
    static func createInstance() -> OPTNotificationCenter? {
        return CustomNotificationCenter()
    }

    func addGenericNotificationListener(notificationType: Int, listener: @escaping GenericListener) {
        //
    }
    
    func removeNotificationListener(notificationId: Int) {
        //
    }
    
    func clearNotificationListeners(type: NotificationType) {
        //
    }
    
    func clearAllNotificationListeners() {
        //
    }
    
    func sendNotifications(type: Int, args: Array<Any?>) {
        //
    }
    
    func addActivateNotificationListener(activateListener: @escaping ActivateListener) {
        //
    }
    
    func addTrackNotificationListener(trackListener: @escaping TrackListener) {
        //
    }
}
